src/datumaro/plugins/transforms.py

```diff
@@ -1837,3 +1845,129 @@ def transform_item(self, item: DatasetItem):
+
+        for col in table.columns:
+            col_data = table[col].dropna()
+            if col in float_cols:
+                skewness = skew(col_data)
+                if abs(skewness) > 1:  # TODO
+                    self._missing_value[col] = col_data.median()
+                else:
+                    self._missing_value[col] = col_data.mean()
+            elif col in countable_cols:
+                self._missing_value[col] = col_data.mode().iloc[0]
+
+    @staticmethod
+    def find_closest_value(series, target_value):
+        abs_diff = np.abs(series - target_value)
+        closest_index = abs_diff.idxmin()
+        return series.iloc[closest_index]
+
+    def cap_outliers(self, table):
+        lower, upper = self._outlier_value[table.name]
+        val = table.iloc[0]
+        if (val < lower) | (val > upper):
+            capped_value = max(min(val, upper), lower)
+            return table.replace(val, capped_value)
+        return val
+
+    def fill_missing_value(self, series):
+        return series.fillna(self._missing_value[series.name])
+
+    def refine_tabular_media(self, item):
+        media = item.media
+        df = pd.DataFrame(media.data(), index=[media.index])
+        str_cols = [col for col in media.data().keys() if item.media.table.dtype(col) is str]
+        float_cols = [col for col in media.data().keys() if item.media.table.dtype(col) is float]
+        int_cols = [col for col in media.data().keys() if item.media.table.dtype(col) is int]
+        countable_cols = [
+            col
+            for col in media.data().keys()
+            if isinstance(item.media.table.dtype(col), CategoricalDtype)
+            or item.media.table.dtype(col) is int
+        ]
+
+        df[str_cols] = df[str_cols].applymap(lambda x: self.remove_unneccessary_char(x))
+
+        if not (self._outlier_value):
+            self.check_outlier(media.table.data[float_cols + int_cols], float_cols + int_cols)
+        df[float_cols + int_cols] = df[float_cols + int_cols].apply(lambda x: self.cap_outliers(x))
+
+        if not (self._missing_value):
+            self.check_missing_value(
+                media.table.data[float_cols + countable_cols], float_cols, countable_cols
+            )
+        df[float_cols + countable_cols] = df[float_cols + countable_cols].apply(
+            lambda x: self.fill_missing_value(x)
+        )
+        # return df.iloc[0].to_dict()
+        return TableRow.from_data(
+            df.iloc[0].to_dict(), table=item.media.table, index=item.media.index
+        )
+
+    def transform_item(self, item):
+        if not isinstance(item.media, TableRow):
+            raise DatumaroError(
+                "Item %s: image info is required for this " "transform" % (item.id,)
+            )
+
+        refined_media = self.refine_tabular_media(item) if item.media.has_data else None
+        refined_annotations = [
+            ann.wrap(values=refined_media.data) if isinstance(ann, Tabular) else ann
```

Comment:
Does this mean that the annotation will be same to refined_media.data? If so, you are thinking about performing `clean` prior to `correct` and `validate`?

Reply:
The approach was planned and implemented as follows:
1. Validate -> Correct: Modify only annotations based on reports, while maintaining media information unchanged.
2. Clean: Modify both media and annotations of dataset items without relying on reports.
If there are any issues with the above approach, please let me know, and I will make the necessary adjustments.

Reply:
Could you double check whether it is okay to `ann.wrap(values=refined_media.data)`?

Reply:
Fix in [59e8ede](https://github.com/openvinotoolkit/datumaro/pull/1520/commits/59e8edeb2a6fde966f5b666dbcf53d3667d93616). I updated values of the annotation with target of dataset.